YARCH/Sources/Modules/CatalogDetails/View/CatalogDetailsView.swift

```diff
@@ -0,0 +1,135 @@
+// Главная вью модуля CatalogDetails
+
+import UIKit
+
+extension CatalogDetailsView {
+    struct Appearance {
+        let tableRowHeight: CGFloat = 60
+        let tableHeaderViewHeight: CGFloat = 150
+    }
+}
+
+class CatalogDetailsView: UIView {
+
+    weak var refreshActionsDelegate: CatalogErrorViewDelegate?
+
+    let appearance: Appearance
+
+    var tableView: UITableView
+
+    var tableHeaderView: CatalogDetailsHeaderView? {
+        return tableView.tableHeaderView as? CatalogDetailsHeaderView
+    }
+
+    lazy var emptyView = CatalogEmptyView()
+
+    lazy var errorView: CatalogErrorView = {
+        let view = CatalogErrorView()
+        view.delegate = self.refreshActionsDelegate
+        return view
+    }()
+
+    init(frame: CGRect = CGRect.zero,
+         loadingDataSource: UITableViewDataSource,
+         loadingDelegate: UITableViewDelegate,
+         refreshDelegate: CatalogErrorViewDelegate, appearance: Appearance = Appearance()) {
+
+        tableView = UITableView(delegate: loadingDelegate, dataSource: loadingDataSource)
+        self.appearance = appearance
+        super.init(frame: frame)
+        refreshActionsDelegate = refreshDelegate
+        configureTableView()
+        addSubviews()
+        makeConstraints()
+
+        emptyView.isHidden = true
+        tableView.isHidden = true
+        errorView.isHidden = true
+    }
+
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    // MARK: Configuration
+
+    func configureHeaderView(_ viewModel: CoinSnapshotFullViewModel) {
+        if let image = viewModel.image {
+            tableHeaderView?.imageView.image = image
+        } else {
+            tableHeaderView?.imageView.becomePlaceholder()
+        }
+    }
+
+    func configureTableView() {
+        tableView.separatorStyle = .none
+        tableView.rowHeight = appearance.tableRowHeight
```

Comment:
Why not use ```UITableViewAutomaticDimension```?

Reply:
Fixed

Reply:
Nope. Not fixed. AL engine magically suggests zero height and assigns default height.